### PR TITLE
Chores #231

### PR DESCRIPTION
**Pull request by mboullouz**
Please take a look at this @garak

### CODE REVIEW

File: src/Knp/Component/Pager/Exception/PagerBadArgumentException.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (garak) [inline src/Knp/Component/Pager/Exception/PagerBadArgumentException.php]: I don't think we need a custom exception

File: src/Knp/Component/Pager/Paginator.php [hunk 2]
```diff
@@ -88,16 +89,15 @@ public function setDefaultPaginatorOptions(array $options): void
...
-     * @throws \LogicException
+     * @throws PagerBadArgumentException
      * @return PaginationInterface
      */
     public function paginate($target, int $page = 1, int $limit = 10, array $options = []): PaginationInterface
     {
-        $limit = intval(abs($limit));
-        if (!$limit) {
-            throw new \LogicException("Invalid item per page number, must be a positive number");
+        if ( $limit<=0 or $page<=0) {
```
Comment (garak) [inline src/Knp/Component/Pager/Paginator.php]: Wrong spacing here: you should omit spaces after/before parentehes and add spaces after/before controls (like `<=`

File: src/Knp/Component/Pager/Paginator.php [hunk 3]
```diff
@@ -88,16 +89,15 @@ public function setDefaultPaginatorOptions(array $options): void
...
+            throw new PagerBadArgumentException("Invalid item per page number. Limit: $limit and Page: $page, must be positive non-zero integers");
         }
-        $offset = abs($page - 1) * $limit;
+        $offset =  ($page - 1) * $limit;
```
Comment (garak) [inline src/Knp/Component/Pager/Paginator.php]: useless space here

File: tests/Test/Pager/Pagination/PaginatorTest.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+final class PaginatorTest extends BaseTestCase
+{
+    /**
+     * @test
+     */
+    public function shouldThrowExceptionOnInvalidPageAndLimitArgs(): void
+    {
+        $exceptionTrown=false;
+        try {
+            (new Paginator( ))->paginate(['a','b'], -1, 0);
```
Comment (garak) [inline tests/Test/Pager/Pagination/PaginatorTest.php]: useless space here

File: tests/Test/Pager/Pagination/PaginatorTest.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        } catch (PagerBadArgumentException $e) {
+            $exceptionTrown =true;
```
Comment (garak) [inline tests/Test/Pager/Pagination/PaginatorTest.php]: missing space here

File: tests/Test/Pager/Pagination/PaginatorTest.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+use Test\Tool\BaseTestCase;
+
+final class PaginatorTest extends BaseTestCase
+{
+    /**
+     * @test
+     */
+    public function shouldThrowExceptionOnInvalidPageAndLimitArgs(): void
+    {
+        $exceptionTrown=false;
```
Comment (garak) [inline tests/Test/Pager/Pagination/PaginatorTest.php]: lack of spaces here

File: src/Knp/Component/Pager/Exception/PagerBadArgumentException.php (hunk 1 above)
Comment (mboullouz) [reply src/Knp/Component/Pager/Exception/PagerBadArgumentException.php]: Yes it's too much for that small case

File: tests/Test/Pager/Pagination/PaginatorTest.php (hunk 6 above)
Comment (mboullouz) [reply tests/Test/Pager/Pagination/PaginatorTest.php]: should be ok in the new commit

File: tests/Test/Pager/Pagination/PaginatorTest.php [hunk 7]
```diff
@@ -2,22 +2,19 @@
 
 namespace Test\Pager\Pagination;
 
-use Knp\Component\Pager\Exception\PagerBadArgumentException;
 use Knp\Component\Pager\Paginator;
 use Test\Tool\BaseTestCase;
 
-final class PaginatorTest extends BaseTestCase
-{
+final class PaginatorTest extends BaseTestCase {
```
Comment (garak) [inline tests/Test/Pager/Pagination/PaginatorTest.php]: Parentheses must be on a new line, as per [PSR-2](https://www.php-fig.org/psr/psr-2/)

File: src/Knp/Component/Pager/Paginator.php [hunk 8]
```diff
@@ -2,6 +2,7 @@
 
 namespace Knp\Component\Pager;
 
+use Knp\Component\Pager\Exception\PagerBadArgumentException;
```
Comment (garak) [inline src/Knp/Component/Pager/Paginator.php]: No more

File: tests/Test/Pager/Pagination/PaginatorTest.php [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+use Knp\Component\Pager\Paginator;
+use Test\Tool\BaseTestCase;
+
+final class PaginatorTest extends BaseTestCase
+{
+    /**
+     * @test
+     */
+    public function shouldThrowExceptionOnInvalidPageAndLimitArgs(): void {
```
Comment (garak) [inline tests/Test/Pager/Pagination/PaginatorTest.php]: Still wrong

**Pull request merged**